dns/dns.go

```diff
@@ -125,3 +238,54 @@ func handleDnsRequest(currentDomain *domain) func(w dns.ResponseWriter, r *dns.M
 		w.WriteMsg(m)
 	}
 }
+
+func Save() (errs []error) {
+	saving.Lock()
+	for _, domain := range domains {
+		if domain.IsV4Changed() {
+			ipv4Data := ""
+			domain.Mutv4.RLock()
+			for key, val := range domain.Ipv4 {
+				ipv4Data += key + "\t" + val + "\n"
+			}
+			domain.Mutv4.RUnlock()
+			err := os.WriteFile(path.Join(currentDataPath, domain.Root+IPV4_FILE), []byte(ipv4Data), 0644)
+			if err != nil {
+				errs = append(errs, err)
+				color.Errorf("Failed to save ipv4 data for domain %s: %s\n", domain.Root, err)
+			} else {
+				// did successfully save, so mark as saved
+				domain.MarkV4Changed(false)
+			}
+		}
+
+		if domain.IsV6Changed() {
+			ipv6Data := ""
+			domain.Mutv6.RLock()
+			for key, val := range domain.Ipv6 {
+				ipv6Data += key + "\t" + val + "\n"
+			}
+			domain.Mutv6.RUnlock()
+			err := os.WriteFile(path.Join(currentDataPath, domain.Root+IPV6_FILE), []byte(ipv6Data), 0644)
+			if err != nil {
+				errs = append(errs, err)
+				color.Errorf("Failed to save ipv6 data for domain %s: %s\n", domain.Root, err)
+			} else {
+				// did successfully save, so mark as saved
+				domain.MarkV6Changed(false)
+			}
+		}
+	}
+	saving.Unlock()
+
+	errLen := len(errs)
+	if errLen > 0 {
+		color.Errorf("%d errors occurred while trying to save\n", errLen)
+	}
+	return
```

Comment:
Useless return statement

Reply:
Not useless, function returns all errors that occurred while saving. Should I remove this?

Reply:
I meant the return statement, not the logic above

Reply:
It said missing return statement when I left it out, that's why I put it there in the first place

Reply:
ok